crates/distribution-types/src/prioritized_distribution.rs

```diff
@@ -300,3 +285,64 @@ impl From<TagCompatibility> for WheelCompatibility {
         }
     }
 }
+
+impl IncompatibleSource {
+    fn is_more_compatible(&self, other: &IncompatibleSource) -> bool {
```

Comment:
Why not `Ord`?

Reply:
I think my branch just made everything `Ord` and then I had, like, `compatibility > existing` or something. But maybe it ends up being more complicated, the logic in here is quite nuanced. (This isn't blocking feedback.)

Reply:
Satisfying `Ord` is actually really challenging. For example, when comparing `RequiresPython` incompatibilities that contain version specifiers we cannot really provide strong ordering. I attempted to do `Ord` at first as well and after much discussion @BurntSushi recommended this approach instead.

I believe the reason this worked on your branch is that you didn't retain information on the incompatibility enum members. Enum members themselves are trivially ordered, but once we store additional metadata about the incompatibility in them we need ordering in a larger hierarchy of types.

If you have other ideas to get around this please share :)

Reply:
Separately.. the nuance here is a little frightening to me. Selecting the "most relevant incompatible distribution" feels like it could be brittle.

Reply:
If there's no clear precedence, can with just stringifying the incompatibility criteria and sort based on that? It's arbitrary yet deterministic and easy to follow.

Reply:
That's actually what I did to implement `Ord` originally, but it's hard to be sure we're meeting all of the requirements for the trait and it seems brittle.

Reply:
I think what you have here is totally fine.